src/components/screens/software-registry-screen.tsx

```diff
@@ -0,0 +1,346 @@
+import {
+  Button,
+  Typography,
+  Grid,
+  CircularProgress,
+  Box,
+  Alert,
+  Container,
+  IconButton,
+  Card,
+} from "@mui/material";
+import { useState, useEffect, useMemo, useRef } from "react";
+import Content from "../software-registry/myContent";
+import strings from "src/localization/strings";
+import { useAtom, useAtomValue } from "jotai";
+import { authAtom } from "src/atoms/auth";
+import { useLambdasApi } from "src/hooks/use-api";
+import { softwareAtom } from "src/atoms/software";
+import type { SoftwareRegistry } from "src/generated/homeLambdasClient";
+import GridViewIcon from "@mui/icons-material/GridView";
+import ListViewIcon from "@mui/icons-material/List";
+import ArrowDownwardIcon from "@mui/icons-material/ArrowDownward";
+import useCreateSoftware from "src/hooks/use-create-software";
+import Recommendations from "../software-registry/Recommendations";
+import AddSoftwareModal from "../software-registry/AddSoftwareModal";
+import Sidebar from "../software-registry/Sidebar";
+
+/**
+ * Software registry screen component
+ */
+const SoftwareScreen = () => {
+  const { softwareApi } = useLambdasApi();
+  const auth = useAtomValue(authAtom);
+  const loggedUserId = auth?.token?.sub ?? "";
+  const [isGridView, setIsGridView] = useState(true);
+  const [applications, setApplications] = useAtom(softwareAtom);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [selectedTags, setSelectedTags] = useState<string[]>([]);
+  const [searchValue, setSearchValue] = useState<string>("");
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [showAll, setShowAll] = useState(false);
+  const recommendationRef = useRef<null | HTMLDivElement>(null);
+  const { createSoftware } = useCreateSoftware(loggedUserId, setApplications);
+
+  /**
+   * Scrolls to the recommendations section.
+   */
+  const scrollToRecommendations = () => {
+    if (recommendationRef.current) {
+      recommendationRef.current.scrollIntoView({ behavior: "smooth" });
+    }
+  };
+
+  /**
+   * Filters the applications based on the logged-in user's id and the application status.
+   *
+   * @returns The list of filtered applications owned by the logged in user.
+   */
+  const myApplications = useMemo(
+    () => applications.filter((app) =>
+      app.users?.includes(loggedUserId) && app.status === "ACCEPTED"
```

Comment:
Again can we replace the uses of status strings that are just strings like here, with the SoftwareStatus enum?